api-scanner/src/main/java/net/corda/plugins/ScanApi.java

```diff
@@ -348,14 +349,10 @@ private Names toNames(Collection<ClassInfo> classes) {
             Map<Boolean, List<String>> partitioned = classes.stream()
                 .map(ClassInfo::toString)
                 .filter(ScanApi::isApplicationClass)
-                .collect(partitioningBy(this::isVisibleAnnotation));
-            /*
-             * These annotations are returned in an order determined by
-             * fast-classpath-scanner. THIS IS BAD! The ordering here
-             * will almost certainly change/break in future versions.
-             * TODO: Order these annotations ourselves!
-             */
-            return new Names(partitioned.get(true), partitioned.get(false));
+                .collect(partitioningBy(this::isVisibleAnnotation, toCollection(LinkedList::new)));
```

Comment:
Why LinkedList? It doesn't look like you're removing elements from the head. In fact you're sorting and so ArrayList would be better anyway.

Reply:
As a rule, I prefer `LinkedList` over `ArrayList` when I don't know how many elements I will be inserting, so that I don't need to resize the array. However, I suppose we can assume that the number of annotations in either list will be "small".